sound/soc/sof/topology.c

```diff
@@ -2137,8 +2136,6 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 	return ret;
 
 err:
```

Comment:
@bardliao do we not need to free sroute and sconnect in case of error with devm_kzalloc? 

If not, I suppose we dont need the "err:' here? we can just return the value when the error occurs?

Reply:
devm_kzalloc is supposed to be free automatically. I suggest that we keep "err:" since there are a lot of goto err in the function. But we can remove the redundant return above the "err:"